File: test/benchmark/fibonacci.lua
local function fib(x)
    if x == 1 or x == 2 then
        return 1
    end
    return fib(x - 1) + fib(x - 2)
end

local st = os.clock()
local result = fib(40)
local et = os.clock()

print('result = ', result, ', ', et - st)

File: test/benchmark/fibonacci.js
import {print} from 'core';

introspect.setUnhandledPromiseRejectionHandler((promise, reason) => {
    introspect.print(`${reason}\n`);
});

function fib(x) {
    if (x == 1 || x == 2) {
        return 1;
    }
    return fib(x - 1) + fib(x - 2);
}

let st = getMillisecondTimeCounter();

let result = fib(40);

let et = getMillisecondTimeCounter();

print(`result = ${result} in ${et - st} ms\n`);
